basket/Basket.js

```diff
@@ -152,6 +152,16 @@ export default class Basket {
     // load default start date instance
     if (!startDateInstance) startDateInstance = this.getCurrentDateInstance()
 
+    // check that selected date and timespan is within available dates
+    const tmpBasketEntry = new BasketEntry({})
+    tmpBasketEntry.setValidFrom(startDateInstance)
+    tmpBasketEntry.setProductDefinitionInstance(definition)
+    const checkResult = this.checkAvailableDates([tmpBasketEntry])
```

Comment:
why does a function to check dates take a basket entry array? does not make sense

Reply:
Doch, ergibt Sinn ;) Es erlaubt nicht bloss einen Basket-Entry zu checken, sondern mehrere. Wird eben genau auf Zeile 228, dein Kommentar unten benötigt. Dadurch kann in der Funktion updateBasketEntries eine for-Schleife gespart werden, was deutlich schöner ist, da Einzeiler.